docs/modules/maintain-cluster/pages/enterprise-rest-api.adoc

```diff
@@ -0,0 +1,431 @@
+
+You can interact with the API using various tools and platforms such as cURL, REST clients (like Postman and Insomnia), or programming languages with HTTP request libraries or built-in support.
+
+The REST API also includes a Swagger page available at http://<host>:<port>/swagger-ui/index.html for making requests.<host> is the server running the REST server and <port> is the rest server port. The Swagger page provides information about each endpoint, including required parameters, request and response structures and types, potential response codes, and example responses. You can use this page to easily navigate and test different API calls directly from the interface.
+
+NOTE: In this document, all the request examples are demonstrated using cURL.
+
+
+== Enabling REST API
+
+WARNING: **hazelcast-enterprise-rest-<VERSION>.jar** should exist at the classpath of the server to use REST API.
+
+REST service is **disabled** in the configuration by default.
+You should enable it on your rest configuration as shown in the below declarative example:
+
+[tabs] 
+==== 
+XML:: 
++ 
+-- 
+[source,xml]
+----
+<hazelcast>
+    <rest enabled="true">
+        ...
+    </rest>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  rest:
+    enabled: true
+----
+====
+
+
+== Changing Default Port
+
+`8443` is the default port for REST API. To change it, you can use the rest configuration as shown in the below declarative example:
+
+[tabs] 
+==== 
+XML:: 
++ 
+-- 
+[source,xml]
+----
+<hazelcast>
+    <rest enabled="true">
+        <port>8080</port>
+    </rest>
+</hazelcast>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast:
+  rest:
+    enabled: true
+    port: 8080
+----
+====
+
+
+== Authentication
+
+=== Configuring Authentication
+
+REST supports simple authentication. To configure authentication in the REST API, you should complete the following steps:
+
+* **#Enable Security:#**
+You need to enable the security to use REST API.
+
+* **#Configure Rest Realm:#**
+You need to introduce a REST realm as the authentication mechanism. This realm will handle user authentication using a simple method based on username and password.
+
+* **#Setup Users and Roles:#**
+You should identify users and their roles within the REST realm.
+
+* **#Configure AccessControlService:#**
+To configure `AccessControlService`, you should provide a factory class that creates an instance of the service. `AccessControlService` delegates authentication to the configured security realm. You can either create a custom factory class, or use the default implementation. For the default implementation, you can use `DefaultAccessControlServiceFactory`, to produce `DefaultAccessControlService`. Authorization decisions in this service are based on three predefined roles: `reader`, who can make requests via GET endpoints only; `writer`, who can make requests including GET, POST, and DELETE; and `admin`, who can make all requests and access admin-specific endpoints.
```

Comment:
I am seeing another step in the REST demo by Isaac (which is defining the `security-realm` under the `rest` configuration). Please see my simplified content suggestion for all the steps above:

. Enable REST
. Define the security realm within the `rest` configuration
. Enable security
. Setup the following under the defined security realm within the `security` configuration:
.. Users and roles
.. Access control service

Reply:
Fixed at https://github.com/hazelcast/hz-docs/pull/1097/commits/c9f1d1641dbe06821715669b42110e4828ebcbf0